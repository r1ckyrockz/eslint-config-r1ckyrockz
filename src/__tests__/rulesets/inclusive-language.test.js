const {
  createInclusiveLanguageRules,
} = require('../../rulesets/inclusive-language');

describe('createInclusiveLanguageRules', () => {
  test('matches snapshot', () => {
    const project = {};

    expect(createInclusiveLanguageRules(project)).toMatchSnapshot();
  });

  test('allows passing custom rules', () => {
    const rule = 'inclusive-language/use-inclusive-words';
    const level = 'off';

    const project = {
      rules: {
        [rule]: level,
      },
    };

    const defaultRuleset = createInclusiveLanguageRules({
      ...project,
      rules: {},
    });

    const result = createInclusiveLanguageRules(project);

    expect(result[rule]).toBe(level);
    expect(result[rule]).not.toBe(defaultRuleset[rule]);

    expect(result).toMatchSnapshot();
  });
});
