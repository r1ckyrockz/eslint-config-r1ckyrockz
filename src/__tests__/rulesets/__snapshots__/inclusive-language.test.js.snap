// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createInclusiveLanguageRules allows passing custom rules 1`] = `
Object {
  "inclusive-language/use-inclusive-words": "off",
}
`;

exports[`createInclusiveLanguageRules matches snapshot 1`] = `
Object {
  "inclusive-language/use-inclusive-words": "error",
}
`;
